### ISSUE TITLE
Fail for transformer_lra_pf32

### ISSUE THREAD

**Issue by nachewigkeit** [facebookresearch/mega#3]
Hello. I noticed that you replicated XFM results on lra, but I failed when training transformer on pathfinder dataset using your default setting: loss was always 1 and acc was always 50%. I've tuned learning rate, weight decay, batch size, but nothing worked. The experiment for mega on Pathfinder works well, and transformer on other dataset also work well. Could you share your command to train transformer on Pathfinder?

Below is the command I used to train on 8 GPUS:

```
model=transformer_lra_pf32
python -u train.py ${DATA} \
    --seed $seed --ddp-backend c10d --find-unused-parameters \
    -a ${model} --task lra-image --input-type image --pixel-normalization 0.5 0.5 \
    --encoder-layers 1 --encoder-embed-dim 128 --encoder-ffn-embed-dim 128 --chunk-size ${CHUNK} \
    --activation-fn 'silu' --attention-activation-fn 'softmax' \
    --norm-type 'batchnorm' --sen-rep-type 'mp' --encoder-normalize-before \
    --criterion lra_cross_entropy --best-checkpoint-metric accuracy --maximize-best-checkpoint-metric \
    --optimizer adam --lr 1e-4 --adam-betas '(0.9, 0.98)' --adam-eps 1e-8 --clip-norm 1.0 \
    --dropout 0.0 --attention-dropout 0.0 --act-dropout 0.0 --weight-decay 0.01 \
    --batch-size 8 --sentence-avg --update-freq 2 --max-update 250000 \
    --lr-scheduler linear_decay --total-num-update 250000 --end-learning-rate 0.0 \
    --warmup-updates 50000 --warmup-init-lr '1e-07' --keep-last-epochs 1 --max-sentences-valid 8 \
    --save-dir ${SAVE} --log-format simple --log-interval 100 --num-workers 0
```

**Comment by XuezheMax** [comment]
Hello,
For replicating Transformer results on LRA, please follow the settings and configurations in the LRA paper. The commands provided in the repo were only for Mega. In fact, it is very hard to train Transformer on Pathfinder and the configuration in LRA paper was the only one we successfully trained it.

**Comment by XuezheMax** [comment]
I need to contact my co-authors who were in charge of replication of Transformer on LRA. Please give me some time.

**Comment by shawnkx** [comment]
This command is old and maybe broken but the hyperparameter should be correct.
```
CUDA_VISIBLE_DEVICES=0,1 python -u train.py ${DATA} \
    --seed $seed --ddp-backend c10d --fp16 \
    -a ${model} --task transformer_lra \
    --best-checkpoint-metric accuracy --maximize-best-checkpoint-metric \
    --optimizer adam --lr 0.0001 --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
    --batch-size 64 --sentence-avg --update-freq 2 \
    --lr-scheduler inverse_sqrt --weight-decay 0. \
    --criterion cross_entropy --max-update 125000 \
    --warmup-updates 4000 --warmup-init-lr '1e-07' \
    --keep-last-epochs 10 --encoder-layers 4 \
    --save-dir ${SAVE} --log-format simple --log-interval 100 --num-workers 0 \
    --tensorboard-logdir ${SAVE} | tee ${SAVE}/log.txt
```
